secure_software_development_fundamentals.md

```diff
@@ -2361,18 +2361,18 @@ Most database systems include a language that can let you create arbitrary queri
 
 Even if the database language is not SQL, if it is an attack on a language for a database system it is often called an SQL injection attack (even though this is technically not accurate). We will focus on SQL, because SQL is very common and once you understand how to counter SQL injection attacks, it is easy to generalize this to any database language.
 
-Here is a trivial example; here is a snippet of Java that tries to do an SQL query:
+Here is a trivial example; here is a snippet of Java that tries to do an SQL query, but does it insecurely::
 
-~~~~
+~~~~java
     String QueryString = "select * from authors where lastname = ' " + search_lastname + " '; ";
-    rs = statement.executeQuery(QueryString);
+    rs = statement.executeQuery(QueryString); // VULNERABLE CODE
 ~~~~
 
 The intent is clear; if **search_lastname** has the value **Fred**, then the database will receive the query “**select * from authors where lastname='Fred';**” - a reasonable SQL query. But remember our warning signs - this code concatenates strings, some of that data is probably provided by an attacker, and we’re doing something called “execute”.  The warning signs are right. Imagine that the attacker provides the input “**Fred’ OR ’a’=’a**”. This will produce the query “**select * from authors where lastname='Fred' OR ’a’=’a’;**” and now the attacker can retrieve the entire database. The attacker could even modify or delete data this way, depending on various factors. This is a simple example of an SQL injection attack; an attacker can insert some characters and inject new or modified commands.
```

Comment:
rm curly quotes

```suggestion
The intent is clear; if **search_lastname** has the value **Fred**, then the database will receive the query “**select * from authors where lastname='Fred';**” - a reasonable SQL query. But remember our warning signs - this code concatenates strings, some of that data is probably provided by an attacker, and we’re doing something called “execute”.  The warning signs are right. Imagine that the attacker provides the input “**Fred' OR 'a'='a**”. This will produce the query “**select * from authors where lastname='Fred' OR 'a'='a';**” and now the attacker can retrieve the entire database. The attacker could even modify or delete data this way, depending on various factors. This is a simple example of an SQL injection attack; an attacker can insert some characters and inject new or modified commands.
```